lib/Driver.cpp

```diff
@@ -63,25 +63,34 @@ std::vector<std::unique_ptr<TestResult>> Driver::Run() {
 
   for (auto &Test : Finder.findTests(Ctx)) {
     auto ObjectFiles = AllObjectFiles();
-    ExecutionResult ExecResult = Runner.runTest(Test.get(), ObjectFiles);
+
+    ExecutionResult ExecResult = Sandbox.run([&](ExecutionResult *SharedResult){
+      *SharedResult = Runner.runTest(Test.get(), ObjectFiles);
+    });
+
     auto BorrowedTest = Test.get();
     auto Result = make_unique<TestResult>(ExecResult, std::move(Test));
 
     for (auto Testee : Finder.findTestees(BorrowedTest, Ctx)) {
       auto ObjectFiles = AllButOne(Testee->getParent());
       for (auto &MutationPoint : Finder.findMutationPoints(MutationOperators, *Testee)) {
+        ExecutionResult R = Sandbox.run([&](ExecutionResult *SharedResult){
+          /// TODO: here the clone of Testee->getParent() will be used very soon instead.
+          /// For now we are applying mutation to the module same as of mutation point.
+          MutationPoint->applyMutation(Testee->getParent());
+
+          auto Mutant = Compiler.CompilerModule(Testee->getParent());
+          ObjectFiles.push_back(Mutant.getBinary());
 
-        /// TODO: here the clone of Testee->getParent() will be used very soon instead.
-        /// For now we are applying mutation to the module same as of mutation point.
-        MutationPoint->applyMutation(Testee->getParent());
+          /// No longer need to revert mutations since forked child dies:
+          /// ~~Rollback mutation once we have compiled the module~~
+          /// MutationPoint->revertMutation();
```

Comment:
At this point  we work only with LLVM IR split into so-called modules.
To run a test the IR need to be compiled into an object file, once we get an object file we can change the module, compile it again and get another object file, i.e. mutated program.
So no state is rolled back.
The idea behind is related to premature optimization: I did the rollback just to not make a copy of a module every time we want to make a mutation. Though the mechanism is not very trivial, so we decided to skip it for now and to move the responsibility for copies to OS: when you do a fork the child inherits all the parent's memory in read-only mode, then when you try to modify parent's memory it implicitly gets copied into child's address space. 
I.e. we get the copying 'for free'. I hope this hypothesis works :)